File: articles/path-dependence-lock-in-and-decision-timing/python/run_all_path_dependence_workflows.py
#!/usr/bin/env python3
"""Run all Python workflows for Path Dependence, Lock-In, and Decision Timing."""

from __future__ import annotations

from pathlib import Path
import subprocess
import sys

ARTICLE_ROOT = Path(__file__).resolve().parents[1]
PYTHON_DIR = ARTICLE_ROOT / "python"

SCRIPTS = [
    "switching_cost_model.py",
    "lock_in_risk_model.py",
    "option_value_decay_model.py",
    "timing_review_model.py",
    "path_strategy_comparison.py",
    "decision_record_exporter.py",
    "path_dependence_lock_in_decision_timing_simulation.py",
]


def main() -> None:
    for script in SCRIPTS:
        path = PYTHON_DIR / script
        print(f"Running {path.name}...")
        subprocess.run([sys.executable, str(path)], check=True)
    print("All Python path-dependence workflows completed.")


if __name__ == "__main__":
    main()

File: articles/path-dependence-lock-in-and-decision-timing/r/run_all_path_dependence_workflows.R
# run_all_path_dependence_workflows.R
# Run all R workflows for Path Dependence, Lock-In, and Decision Timing.

args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)

if (length(file_arg) > 0) {
  script_path <- normalizePath(sub("^--file=", "", file_arg[1]), mustWork = TRUE)
  article_root <- normalizePath(file.path(dirname(script_path), ".."), mustWork = TRUE)
} else {
  article_root <- getwd()
}

r_dir <- file.path(article_root, "r")

scripts <- c(
  "path_dependence_lock_in_decision_timing_workflow.R",
  "path_profiles.R",
  "scenario_performance.R",
  "lock_in_review_tables.R",
  "timing_summary.R"
)

for (script in scripts) {
  path <- file.path(r_dir, script)
  message("Running ", path)
  source(path)
}

message("All R path-dependence workflows completed.")
